dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestLineInterface.cpp

```diff
@@ -185,5 +185,17 @@ TEST(DataInterfaceTest, Line) {
   ASSERT_DOUBLE_EQ(li2.getQ1(), 0.0);
   ASSERT_DOUBLE_EQ(li2.getP2(), 0.0);
   ASSERT_DOUBLE_EQ(li2.getQ2(), 0.0);
+
+  MySecondLine.getTerminal1().setP(std::nan("not a number"));
+  ASSERT_DOUBLE_EQ(li2.getP1(), 0.0);
+
+  int i=2; int j=4*i; j+=2;
```

Comment:
No need for such complexity, you can use double val = 4;

Reply:
Same as above, I'm not sure to understand what to test. The test with a double is already done on above lines like "MyLine.getTerminal1().setP(999.999L);" and then getP().

Here I'm trying to test to set P to nan and/or infinite. The complexity is only here so compilers don't detect a zero divide.

Reply:
Why not using the std::numeric_limits<double>::infinity()?
Moreover please also add a test with finite values to make sure that we still get 0.

Reply:
I did not know about std::numeric_limits::infinity. Thx for the hint :)
I may understand what to test now. It lacks a test that getP returns 0.0 when disconnected, right ? And whatever value we assign to P right ?

Reply:
yes please.

Reply:
done :)